### ISSUE TITLE
Allow any invoccable with the parameter `glz::manage_state` in `glz::manage`

### ISSUE THREAD

**Issue by sam20908** [stephenberry/glaze#480]
Currently, the restriction is that the `handle_*` functions must exist within the structure. This is unnecessary because in some cases, we want to use `glz::manage` to execute some code after reading the JSON data (if there is an easier way in the library, please let me know).

My proposal is that we allow any function that takes in `glz::manage_state` as the parameter. Since this library is using C++20, we can pass a lambda as a template parameter, which we can pass a lambda that takes a reference to the struct itself.

**Comment by stephenberry** [comment]
I agree that `glz::manage` is currently too restrictive, and I think it might be better to simply make `glz::custom` more powerful to be able to handle `glz::manage` use cases. I'm experimenting with adding support for lambdas in `glz::custom`, which I think will allow the same logic as `glz::manage` was designed for.

There are already multiple ways to handle [custom serialization](https://github.com/stephenberry/glaze/blob/main/docs/custom-serialization.md), so I think I'd prefer to eliminate `glz::manage` if possible.

**Comment by sam20908** [comment]
> so I think I'd prefer to eliminate glz::manage if possible.

`glz::custom` can also be used to parse data into another form without requiring another data member and `glz::manage`, but (I don't know if this is correct) it'll have to parse the said value twice because once to read the entire value string and pass it to the callback, and another to parse the said string into some intermediate form. `glz::manage` doesn't have this problem because it simply calls the function after the value is parsed and constructed. 
TL;DR: They are not exactly the same, so it may be best to keep `glz::manage`.

**Comment by stephenberry** [comment]
Yes, when I was updated `glz::manage` I realized it could solve problems that `glz::custom` could not. So, I'm definitely keeping it and have a use case with a hardware interface where I'll be using it. Thanks for the feedback!
